### DISCUSSION TITLE
stash is not a valid reference

### DISCUSSION THREAD

**Discussion by Lilweavs** [jesseduffield/lazygit#2786] [General]
I have been migrating to lazygit and out of the blue my stashes are no longer working. I can create a stash, but nothing shows up if I want to view the changes stored in the stash.

If I try to pop/apply the stash I get error: stash@0 is not a valid reference. I have been using lazygit for several weeks now and started happening suddenly. If I manually run `git stash pop` from the command line everything works fine. 

Any ideas on what could be happening?

**Comment by mark2185** [comment]
Could you fill out a bug report? Just so we have more information.

**Comment by Lilweavs** [comment]
**Description**
When applying/popping a stash, lazygit reports `error: stash@# is not a valid stash`. I can manually pop the stash via `git stash pop/apply` and the file that I stashed shows up as expected. Additionally, the changes in a stash are no longer viewable.

**To Reproduce**
Steps to reproduce the behavior:

1. Modify any file
2. Stash file via: `s` with any name
3. Go to stash tab and `enter` on stash created at step 2
4. `Esc` and apply/pop stash

**Expected behavior**
At step 3 I would expect to be able to view the changes in stash (nothing shows). At step 4 I would expect the stash to either be applied or popped.

**Screenshots**
![image](https://github.com/jesseduffield/lazygit/assets/35853304/0b724c23-b87b-4eb5-b311-1e17b71f83c6)
![image](https://github.com/jesseduffield/lazygit/assets/35853304/a8900545-15ce-4cd8-80b3-478f43c316e3)


**Version info:**
`commit=5149b24ab3dfad3860e2300519c7c583dcc8c9ff, build date=2023-05-03T08:00:22Z, build source=binaryRelease, version=0.38.2, os=windows, arch=amd64, git version=2.40.1`
`git version 2.40.1`


**Additional context**
Step 3 and 4 were working as expected earlier today. I am running on Windows 11 with Windows terminal.
